reservation_units/models/reservation_unit.py

```diff
@@ -268,23 +272,228 @@ class Meta:
+                    & models.Q(publish_ends__isnull=False)
+                    & models.Q(publish_ends__gt=Now())
+                    # Publishes before it unpublishes.
+                    & models.Q(publish_begins__lt=models.F("publish_ends"))
+                ),
+                then=models.Value(ReservationUnitPublishingState.SCHEDULED_PERIOD.value),
+            ),
+            # Otherwise, Reservation Unit is published.
+            default=models.Value(ReservationUnitPublishingState.PUBLISHED.value),
+            output_field=models.CharField(),
+        )
+        return case  # type: ignore[return-value]  # noqa: RET504
+
+    @lookup_property
+    def reservation_state() -> ReservationUnitReservationState:
+        case = models.Case(
+            # Reservation Unit is currently not reservable, but will be in the future.
+            models.When(
+                (
+                    # Reservation period begins in the future.
+                    models.Q(reservation_begins__isnull=False)
+                    & models.Q(reservation_begins__gt=Now())
+                    & (
+                        # No previous reservation period.
+                        models.Q(reservation_ends__isnull=True)
+                        # Previous reservation period ended in the past.
+                        | models.Q(reservation_ends__lte=Now())
+                    )
+                ),
+                then=models.Value(ReservationUnitReservationState.SCHEDULED_RESERVATION.value),
+            ),
+            # Reservation Unit is currently not reservable, but will be in the future for a specific period.
+            models.When(
+                (
+                    # Reservation period begins in the future.
+                    models.Q(reservation_begins__isnull=False)
+                    & models.Q(reservation_begins__gt=Now())
+                    # Reservation period ends in the future
+                    & models.Q(reservation_ends__isnull=False)
+                    & models.Q(reservation_ends__gt=Now())
+                    # Reservation period begins before it ends.
+                    & models.Q(reservation_begins__lt=models.F("reservation_ends"))
+                ),
+                then=models.Value(ReservationUnitReservationState.SCHEDULED_PERIOD.value),
+            ),
+            # Reservation Unit doesn't have an active pricing,
+            #  OR it has a paid pricing with no payment product,
+            #  OR it was reservable in the past (but not anymore),
+            #  OR reservation period is zero-length.
+            models.When(
+                (
+                    # No active pricing.
+                    L(active_pricing_type=None)
+                    | (
+                        # Paid pricing with no payment product.
+                        L(active_pricing_type=models.Value(PricingType.PAID.value))
+                        & models.Q(payment_product__isnull=True)
```

Comment:
is the check for.MOCK_VERKKOKAUPPA_API_ENABLED no longer necessary?

Reply:
It is necessary, I just forgot to add it back in. Will fix on Monday.

Reply:
Done.